week3/main.cpp

```diff
@@ -0,0 +1,130 @@
+//CodeU: Assignment_3
+//
+//  Created by Julia Semavina on 8.06.2017
+//
+//  problem specification:
+//      Given a grid of letters and a dictionary, find all the words from the dictionary that can be
+//      formed in the grid.
+//
+//  idea:
+//      We start building words from every cell of the grid. So, at first we have words of the length=1. If the word
+//      is prefix, we add one letter (from the North, South, East, West, N-W, N-E, S-W, S-E) if it is possible. Now we
+//      have words of the length=2.
+//      But this way if the input grid consists of the same items and input dictionary contains a very long string
+//      of the same item (like "aaaaaaaa...aaa"), the complexity is SizeOfGrid*(C^MaxLength).
+//      To avoid this we have SamePaths, which is map<int index_in_grip, string word>. So if in some case we find
+//      path, which pair of index_in_grip and word already is in SamePath, we do not continue this path.
+//
+//  comments:
+//      register is ignored
+//
+
+
+#include <iostream>
+#include <set>
+#include <map>
+#include "word.h"
+
+
+void WordsFromPrefix(int index, std::string prefix, int NumberOfColumns, int NumberOfRows, char * Grid,
+                     Dictionary &TheDictionary,
+                     std::set<std::string> &Answer, std::set<std::pair<int, std::string>> &SamePaths) {
```

Comment:
Too many parameters. Maybe it would be better to wrap this function and function below into class?